File: week_3_dp/design-pattern/StrategyFactoryTemplate/after/Factory.ts
import { Handler, PersonA } from "./Handler";

// 工廠設計模式
export class Factory {
  private static strategyMap: Map<string, Handler> = new Map();

  static getStrategy(name: string): Handler {
    const strategy = Factory.strategyMap.get(name);
    if (strategy) {
      return strategy;
    } else {
      throw new Error("不存在");
    }
  }
  static register(name: string, handler: Handler) {
    if (handler == null) {
      return;
    }
    this.strategyMap.set(name, handler);
  }
}

File: week_3_dp/java-dp/src/main/java/com/example/demo/StrategyFactoryTemplate/After/Factory.java
package com.example.demo.StrategyFactoryTemplate.After;

import org.springframework.util.StringUtils;
import org.testng.collections.Maps;

import java.util.Map;

public class Factory {
    private static final Map<String, Handler> strategyMap = Maps.newHashMap();

    public static Handler getStrategy(String str) {
        return strategyMap.get(str);
    }

    public static void register(String name, Handler handler) {
        if (!StringUtils.hasText(name) || null == handler) {
            return;
        }
        strategyMap.put(name, handler);
    }
}
